modules/mobile/spec/request/prescriptions_request_spec.rb

```diff
@@ -0,0 +1,280 @@
+          expect(response.parsed_body['links']).to eq(
+            {
+              'self' =>
+                'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=2&filter[[refill_status][eq]]=refillinprocess&sort=-refill_date',
+              'first' =>
+                'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=1&filter[[refill_status][eq]]=refillinprocess&sort=-refill_date',
+              'prev' =>
+                'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=1&filter[[refill_status][eq]]=refillinprocess&sort=-refill_date',
+              'next' =>
+                'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=3&filter[[refill_status][eq]]=refillinprocess&sort=-refill_date',
+              'last' =>
+                'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=3&filter[[refill_status][eq]]=refillinprocess&sort=-refill_date'
+            }
+          )
+
+          statuses = response.parsed_body['data'].map { |d| d.dig('attributes', 'refillStatus') }.uniq
+          expect(statuses).to eq(['refillinprocess'])
+
+          expect(response.parsed_body['data'].map { |p| p.dig('attributes', 'refillDate') }).to eq(
+            %w[2017-01-25T05:00:00.000Z 2016-11-30T05:00:00.000Z 2016-11-30T05:00:00.000Z]
+          )
+        end
+      end
+    end
+
+    context 'filtering parameters' do
+      context 'filter by refill status' do
+        let(:filter_param) { 'filter[[refill_status][eq]]=refillinprocess' }
+
+        it 'returns all prescriptions that are refillinprocess status', :aggregate_failures do
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get "/mobile/v0/health/rx/prescriptions?#{filter_param}", headers: iam_headers
+          end
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to match_json_schema('prescription')
+
+          refill_statuses = response.parsed_body['data'].map { |d| d.dig('attributes', 'refillStatus') }.uniq
+          expect(refill_statuses).to eq(['refillinprocess'])
+        end
+      end
+
+      context 'filter by not equal to refill status' do
+        let(:filter_param) { 'filter[[refill_status][not_eq]]=refillinprocess' }
+
+        it 'returns all prescriptions that are not refillinprocess status', :aggregate_failures do
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get "/mobile/v0/health/rx/prescriptions?#{filter_param}", headers: iam_headers
+          end
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to match_json_schema('prescription')
+
+          refill_statuses = response.parsed_body['data'].map { |d| d.dig('attributes', 'refillStatus') }.uniq
+          expect(refill_statuses).not_to include('refillinprocess')
```

Comment:
good point, would the best way to handle it be to have another `expect` that checks if `refill_statuses` is > 0?

Reply:
Overall, I don't think this is a big deal. When I was reading it, I just couldn't help but notice that this is asserting a negative and, because of the way dig works, it's a fragile negative test. It would be very easy to make a typo or change the underlying code without breaking this test. You could assert for length or just do the positive assertion:
`expect(refill_statuses).to eq(['foo', 'bar']) # does not include refillinprocess`